src/providers/fail_over.c

```diff
@@ -488,13 +488,14 @@ service_destructor(struct fo_service *service)
 static bool fo_is_ip_address(char *name)
 {
     struct sockaddr_in sa;
+    struct sockaddr_in6 sa6;
```

Comment:
Why allocating two whole `sockaddr_in(6)`  structures and then use just the `sin(6)_addr` field? Wouldn't it be better to allocate `in_addr` and `in6_addr` structures?

Reply:
Hmm, it is not crucial but you are right. We even do not need two of them. Just the bigger one.

Reply:
> Just the bigger one.

Please keep both, size is (glibc) implementation detail.

Reply:
@aplopez what about this update

Reply:
@alexey-tikhonov I have the original suggestion saved. Just take a look.

Reply:
I would keep two structs - `in6_addr` and `in_addr`.

Reply:
I think a comment stating why you are using the same structure for both will help understand why you did this.

Reply:
I think that having two structures make it more readable. I just changed it to use the minimal structure as @aplopez suggested originally.